src/main.rs

```diff
@@ -95,10 +94,57 @@ fn test_list_sql_files() -> io::Result<()> {
 		PathBuf::from("migrations/30_yo.sql"),
 	]);
 
-	purge_migrations_directory(DEFAULT_MIGRATIONS_DIRECTORY)?;
+	purge_directory(DEFAULT_MIGRATIONS_DIRECTORY)?;
 	Ok(())
 }
 
+#[test]
+#[serial_test::serial]
+fn test_list_sql_files_nested_schema() -> io::Result<()> {
+    use pretty_assertions::assert_eq;
+
+    purge_directory(&DEFAULT_SCHEMA_DIRECTORY)?;
+    ensure_directory(&DEFAULT_SCHEMA_DIRECTORY)?;
+
+    fs::File::create("schema/README")?;
+    fs::File::create("schema/00_types.sql")?;
+    fs::create_dir("schema/01_tables")?;
+    fs::File::create("schema/01_tables/00_foo.sql")?;
+    fs::create_dir("schema/01_tables/01_bar_tables")?;
+    fs::File::create("schema/01_tables/01_bar_tables/README")?;
+    fs::File::create("schema/01_tables/01_bar_tables/bar_0.sql")?;
+    fs::File::create("schema/01_tables/01_bar_tables/bar_1.sql")?;
+    fs::create_dir("schema/02_functions")?;
+    fs::File::create("schema/02_functions/00_common.sql")?;
+    fs::create_dir("schema/02_functions/01_bar_functions")?;
+    fs::File::create("schema/02_functions/01_bar_functions/00_get_bar.sql")?;
+    fs::File::create("schema/02_functions/01_bar_functions/01_aggregate_bar.sql")?;
+    fs::create_dir("schema/02_functions/02_foo_functions")?;
+    fs::File::create("schema/02_functions/02_foo_functions/README")?;
+    fs::File::create("schema/02_functions/02_foo_functions/00_get_foo.sql")?;
+    fs::File::create("schema/02_functions/02_foo_functions/01_aggregate_foo.sql")?;
+    fs::File::create("schema/03_indexes.sql")?;
+
+    let schema_files = list_sql_files(&DEFAULT_SCHEMA_DIRECTORY)?;
+    assert_eq!(
+        schema_files,
+        vec![
+            PathBuf::from("schema/00_types.sql"),
+            PathBuf::from("schema/01_tables/00_foo.sql"),
+            PathBuf::from("schema/01_tables/01_bar_tables/bar_0.sql"),
+            PathBuf::from("schema/01_tables/01_bar_tables/bar_1.sql"),
+            PathBuf::from("schema/02_functions/00_common.sql"),
+            PathBuf::from("schema/02_functions/01_bar_functions/00_get_bar.sql"),
+            PathBuf::from("schema/02_functions/01_bar_functions/01_aggregate_bar.sql"),
+            PathBuf::from("schema/02_functions/02_foo_functions/00_get_foo.sql"),
+            PathBuf::from("schema/02_functions/02_foo_functions/01_aggregate_foo.sql"),
+            PathBuf::from("schema/03_indexes.sql"),
+        ]
+    );
+
+    purge_directory(&DEFAULT_SCHEMA_DIRECTORY)?;
+    Ok(())
+}
```

Comment:
Can you add an example (or change this example) to have files and directories with the same name (`00_base.sql` and `00_base/etc`). I understand that won't be what most people will do, but I want to make sure we understand and can point people to the right order if they happen to do that or have reasons to do so.

Reply:
I think I've updated this how you described, but let me know if you'd like anything else done here.